lib/rubocop/cop/style/object_then.rb

```diff
@@ -33,39 +33,37 @@ class ObjectThen < Base
         RESTRICT_ON_SEND = %i[then yield_self].freeze
 
         def on_block(node)
+          return unless RESTRICT_ON_SEND.include?(node.method_name)
+
           check_method_node(node.send_node)
         end
-
         alias on_numblock on_block
 
         def on_send(node)
           return unless node.arguments.one? && node.first_argument.block_pass_type?
 
           check_method_node(node)
         end
+        alias on_csend on_send
 
         private
 
         def check_method_node(node)
-          return unless preferred_method?(node)
-
-          message = message(node)
-          add_offense(node.loc.selector, message: message) do |corrector|
-            prefer = style == :then && node.receiver.nil? ? 'self.then' : style
+          if preferred_method?(node)
+            correct_style_detected
+          else
+            opposite_style_detected
```

Comment:
`correct_style_detected` and `opposite_style_detected` were not being called previously but I think they belong.